modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/repository/RepositoryDistributionSnapshot.java

```diff
@@ -434,4 +444,97 @@ public DistributionSnapshot readJson(InputStream in) {
+            props.put(PROP_RELEASED, null);
+        } else {
+            props.put(PROP_RELEASED, new SimpleDateFormat("yyyy-MM-dd").format(releaseDate));
+        }
+        List<String> aliases = safeGet(PROP_ALIASES);
+        props.put(PROP_ALIASES, String.join("\n", aliases));
+        return props;
+    }
+
+    /**
+     * Returns a key/value map of properties for update from request proprties.
+     *
+     * @since 11.2
+     */
+    public Map<String, String> getUpdateProperties(Map<String, String[]> formFields) {
+        Map<String, String> props = new HashMap<>();
+        if (formFields != null) {
+            Stream.of(TITLE_PROPERTY_PATH, PROP_NAME, PROP_VERSION, PROP_KEY, PROP_LATEST_LTS, PROP_LATEST_FT,
+                    PROP_HIDE)
+                  .filter(entry -> formFields.containsKey(entry))
+                  .forEach(p -> props.put(p, formFields.get(p)[0]));
+            if (StringUtils.isBlank(TITLE_PROPERTY_PATH)) {
+                props.put(TITLE_PROPERTY_PATH, props.get(PROP_NAME));
+            }
+            Stream.of(PROP_LATEST_LTS, PROP_LATEST_FT, PROP_HIDE)
+                  .forEach(p -> props.put(p, Boolean.toString(formFields.containsKey(p))));
+            if (formFields.containsKey(PROP_RELEASED)) {
+                props.put(PROP_RELEASED, formFields.get(PROP_RELEASED)[0]);
+            }
+            if (formFields.containsKey(PROP_ALIASES)) {
+                props.put(PROP_ALIASES, formFields.get(PROP_ALIASES)[0]);
+            }
+        }
+        return props;
+    }
+
+    public DocumentModel updateDocument(CoreSession session, Map<String, String> updateProperties, String comment)
+            throws DocumentValidationException {
+        DocumentModel doc = getDoc();
+        if (updateProperties != null) {
+            if (Stream.of(TITLE_PROPERTY_PATH, PROP_NAME, PROP_VERSION, PROP_KEY)
+                      .anyMatch(p -> StringUtils.isBlank(updateProperties.get(p)))) {
+                throw new DocumentValidationException("Please fill all required fields.");
+            }
+            Stream.of(TITLE_PROPERTY_PATH, PROP_NAME, PROP_VERSION, PROP_KEY, PROP_LATEST_LTS, PROP_LATEST_FT,
+                    PROP_HIDE)
+                  .filter(entry -> updateProperties.containsKey(entry))
+                  .forEach(p -> doc.setPropertyValue(p, updateProperties.get(p)));
+            Stream.of(PROP_LATEST_LTS, PROP_LATEST_FT, PROP_HIDE)
+                  .forEach(p -> doc.setPropertyValue(p, updateProperties.get(p)));
+            if (updateProperties.containsKey(PROP_RELEASED)) {
+                String released = updateProperties.get(PROP_RELEASED);
+                if (StringUtils.isNotBlank(released)) {
+                    LocalDate date = LocalDate.parse(released);
+                    Instant instant = date.atStartOfDay().atZone(ZoneId.systemDefault()).toInstant();
+                    doc.setPropertyValue(DistributionSnapshot.PROP_RELEASED, java.util.Date.from(instant));
```

Comment:
`java.util.Date` is imported by the class, the full qualified name shouldn't be needed.